code/edge/src/test/java/com/adobe/marketing/mobile/EdgeHitProcessorTests.java

```diff
@@ -1287,6 +1216,260 @@ public void setLocationHint(final String hint, final int ttlSeconds) {
+
+		ArgumentCaptor<String> urlCaptor = ArgumentCaptor.forClass(String.class);
+		ArgumentCaptor<String> payloadCaptor = ArgumentCaptor.forClass(String.class);
+		verify(mockEdgeNetworkService, times(1))
+			.doRequest(
+				urlCaptor.capture(),
+				payloadCaptor.capture(),
+				ArgumentMatchers.anyMap(),
+				any(EdgeNetworkService.ResponseCallback.class)
+			);
+
+		// Assert that the url contains the overridden datastreamId
+		assertTrue(urlCaptor.getValue().contains("configId=testDatastreamId"));
+
+		assertTrue(payloadCaptor.getValue().contains("sdkConfig"));
+
+		JSONObject requestJson = new JSONObject(payloadCaptor.getValue());
+		assertNotNull(requestJson);
+		JSONObject metaJson = requestJson.getJSONObject("meta");
+		assertNotNull(metaJson);
+		JSONObject sdkConfigJson = metaJson.getJSONObject("sdkConfig");
+		assertNotNull(sdkConfigJson);
+		JSONObject datastreamJson = sdkConfigJson.getJSONObject("datastream");
+		assertNotNull(datastreamJson);
+
+		assertEquals("works", datastreamJson.get("original"));
+	}
+
+	@Test
+	public void testProcessHit_experienceEvent_whenDatastreamConfigOverridesSet_sendsConfigOverrides()
+		throws Exception {
+		mockNetworkServiceResponse("https://test.com", new RetryResult(EdgeNetworkService.Retry.NO));
+
+		Map<String, Object> configOverrides = new HashMap<>();
+		configOverrides.put("key", "value");
+
+		DataEntity dataEntity = new EdgeDataEntity(
+			getExperienceEventWithConfig(null, configOverrides),
+			edgeConfig,
+			identityMap
+		)
+			.toDataEntity();
+		// test
+		hitProcessor =
+			new EdgeHitProcessor(
+				mockNetworkResponseHandler,
+				mockEdgeNetworkService,
+				mockNamedCollection,
+				mockSharedStateCallback,
+				new EdgeStateCallback() {
+					@Override
+					public Map<String, Object> getImplementationDetails() {
+						return implementationDetails;
+					}
+
+					@Override
+					public String getLocationHint() {
+						return null;
+					}
+
+					@Override
+					public void setLocationHint(final String hint, final int ttlSeconds) {
+						// not called by hit processor
+					}
+				}
+			);
+		assertProcessHitResult(dataEntity, true);
+
+		ArgumentCaptor<String> payloadCaptor = ArgumentCaptor.forClass(String.class);
+		verify(mockEdgeNetworkService, times(1))
+			.doRequest(
+				anyString(),
+				payloadCaptor.capture(),
+				ArgumentMatchers.anyMap(),
+				any(EdgeNetworkService.ResponseCallback.class)
+			);
+
+		assertTrue(payloadCaptor.getValue().contains("configOverrides"));
+
+		JSONObject requestJson = new JSONObject(payloadCaptor.getValue());
+		assertNotNull(requestJson);
+		JSONObject metaJson = requestJson.getJSONObject("meta");
+		assertNotNull(metaJson);
+		JSONObject configOverridesJson = metaJson.getJSONObject("configOverrides");
+		assertNotNull(configOverridesJson);
+
+		assertEquals("value", configOverridesJson.get("key"));
+	}
+
+	@Test
+	public void testProcessHit_experienceEvent_whenDatastreamIdAndConfigOverridesSet_sendsSdkConfigAndConfigOverrides()
```

Comment:
@addb I see an overlap between unit and functional tests especially on the request payload, can you summarize what is being tested and what level?

Reply:
I though of adding these unit tests since hit processor is responsible for adding the overrides payload and also passes datastream Id for the URL. So it felt like adding tests to verify those things here, which I agree would also be captured in the functional tests.

Reply:
Unit tests is a great candidate for the heavy testing for good coverage, and on top of that we should minimize the amount of test duplication at functional/integration testing. 
@addb @kevinlind what test schema (unit > functional > integration) do you recommend for this feature?

Reply:
@emdobrin @kevinlind 
Should I add the functional test in this PR, so it helps with the decision about @emdobrin's comment, as to what to test where?

Reply:
I think we should keep all the unit tests. We can reduce the functional test and integration test to 1 and test everything(datastream id override + datastream config override) in that one test. Unit tests are fast and whereas functional and integration tests are slow and would increase build time as well.

Reply:
Thanks for all the updates on the tests @addb , looks 💯 !